File: src/test/java/br/com/blz/testjava/product/model/ProductTest.java
package br.com.blz.testjava.product.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Set;

import javax.validation.ConstraintViolation;
import javax.validation.Validation;
import javax.validation.Validator;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class ProductTest {

    private Validator validator;
    
    private Product defaultProduct; 

    @BeforeEach
    public void setupValidatorInstance() {
        validator = Validation.buildDefaultValidatorFactory().getValidator();
        defaultProduct = Product.builder().sku(123456L).name("L'oreal").build();
    }

	@Test
	void mustNotLetCreateProductWithNoSKU() {
		defaultProduct.setSku(null);
		
		Set<ConstraintViolation<Product>> violations = validator.validate(defaultProduct);
		
		assertEquals(violations.size(), 1);
		
	}

	@Test
	void mustNotLetCreateProductWithoutName() {
		defaultProduct.setName(null);
		
		Set<ConstraintViolation<Product>> violations = validator.validate(defaultProduct);
		
		assertEquals(violations.size(), 1);
		
	}

}

File: src/test/java/br/com/blz/testjava/product/model/ProductTest.kt
package br.com.blz.testjava.product.model

import br.com.blz.testjava.inventory.provider.InventoryProvider
import br.com.blz.testjava.product.service.provider.ProductProvider
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.junit.jupiter.MockitoExtension

@ExtendWith(MockitoExtension::class)
internal class ProductTest {
  @Test
  fun `should return isMarketable true when inventory has quantity`() {
    val product = ProductProvider.createEntity(InventoryProvider.createEntityWithATotalOf15WareHouse())
    assertTrue(product.isMarketable)
  }

  @Test
  fun `should return isMarketable false when inventory has no quantity`() {
    val product = ProductProvider.createEntity(InventoryProvider.createEntityWithoutWareHouse())
    assertFalse(product.isMarketable)
  }
}
